maas-api/internal/handlers/models.go

```diff
@@ -39,9 +40,37 @@ func (h *ModelsHandler) ListModels(c *gin.Context) {
 
 // ListLLMs handles GET /v1/models.
 func (h *ModelsHandler) ListLLMs(c *gin.Context) {
-	modelList, err := h.modelMgr.ListAvailableLLMs(c.Request.Context())
+	// Extract user context from middleware
+	userCtx, exists := c.Get("user")
+	if !exists {
+		c.JSON(http.StatusInternalServerError, gin.H{
+			"error": gin.H{
+				"message": "User context not found",
+				"type":    "server_error",
+			}})
+		return
+	}
+
+	user, ok := userCtx.(*token.UserContext)
+	if !ok {
+		c.JSON(http.StatusInternalServerError, gin.H{
+			"error": gin.H{
+				"message": "Invalid user context type",
+				"type":    "server_error",
+			}})
+		return
+	}
+
+	// Extract the SA token from the Authorization header
+	authHeader := c.GetHeader("Authorization")
+	saToken := ""
+	if authHeader != "" && len(authHeader) > 7 { // "Bearer " = 7 chars
+		saToken = authHeader[7:] // Remove "Bearer " prefix
+	}
```

Comment:
How this will work when I pass my user token which I think is generally the case for listing the model instead of SA token (or API key if we would be using those)?

Maybe we can also use `CutPrefix("Bearer")`  + `TrimSpace` to avoid magic numbers.